blockchain_connector/Dockerfile

```diff
@@ -120,53 +115,139 @@ RUN mkdir -p build \
+
+COPY VERSION /project/avalon/
+COPY ./bin /project/avalon/bin
+COPY ./sdk /project/avalon/sdk
+
+ENV TCF_HOME=/project/avalon
 
 WORKDIR /project/avalon/sdk
-#install Avalon SDK package.
+
+#install Avalon SDK  package.
 RUN echo "Building Avalon SDK\n" \
  && make
 
 
-# Install fabric python sdk
-RUN pip3 install virtualenv \
-   && pip3 install git+https://github.com/hyperledger/fabric-sdk-py.git
-
-# install blockchain connector module
-RUN cd /project/avalon/blockchain_connector \
-   && make
-
-# Build final image and install modules
+# Build Avalon blockchain connector docker image
 FROM base_image as final_image
+# Add necessary packages to build
+RUN apt-get update \
+ && apt-get install -y -q \
+    pkg-config \
+    protobuf-compiler \
+    libprotobuf-dev \
+    software-properties-common \
+    dh-autoreconf \
+    ocaml \
+    xxd \
+    ocamlbuild \
+    cmake \
+    make \
+    git \
+    swig \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
 
-ENV TCF_HOME=/project/avalon
+RUN apt-get update \
+ && apt-get install -y -q python3-pip \
+ && pip3 install setuptools json-rpc web3 py-solc-x nose2 \
+ && pip3 install nest_asyncio \
+ && echo "Install fabric python sdk\n" \
+ && pip3 install virtualenv \
+ && pip3 install git+https://github.com/hyperledger/fabric-sdk-py.git \
+ && apt-get autoremove --purge -y -q python3-pip
+
+# Install specific version of solc to support versions <0.6.0
+RUN python3 -m solcx.install v0.5.15
 
 WORKDIR /project/avalon/blockchain_connector
 
-COPY --from=build_image /usr/local/ /usr/local
-COPY --from=build_image /root/.solcx /root/.solcx
+COPY VERSION /project/avalon/
+COPY ./bin /project/avalon/bin
+COPY ./blockchain_connector /project/avalon/blockchain_connector
+
 # Copy Python build artifacts
-COPY --from=build_image /project/avalon/common/python/dist/*.whl dist/
-COPY --from=build_image /project/avalon/common/crypto_utils/dist/*.whl dist/
-COPY --from=build_image /project/avalon/sdk/dist/*.whl dist/
-COPY --from=build_image /project/avalon/blockchain_connector/dist/*.whl dist/
-
-# Installing wheel file requires python3-pip package.
-# python3-pip package will increase size of final docker image.
-# So remove python3-pip package and dependencies.
+COPY --from=common_python_image /project/avalon/common/python/dist/*.whl dist/
+COPY --from=common_crypto_utils_image /project/avalon/common/crypto_utils/dist/*.whl dist/
+COPY --from=avalon_sdk_image /project/avalon/sdk/dist/*.whl dist/
+
+ENV TCF_HOME=/project/avalon
+
+# But python3-pip package will increase size of final docker image.
+# So remove python3-pip package and dependencies after installing wheel file.
 RUN apt-get update \
  && apt-get install -y -q python3-pip \
- && echo "Install Common Python, Crypto, SDK & Blockchain_connector packages\n" \
+ && echo "Install Common Python and SDK packages\n" \
+ && pip3 install dist/*.whl \
+ && echo "Generate and install blockchain connector module\n" \
+ && python setup.py bdist_wheel \
```

Comment:
I added a makefile explicity to make the build process uniform for all modules. Should we use it here?

Reply:
I see. Yes, I am going to address this in this PR.